src/figuresClasses.ts

```diff
@@ -1,19 +1,86 @@
-export interface Figure {
+type FigureColor = 'red' | 'green' | 'blue';
+type FigureShape = 'triangle' | 'circle' | 'rectangle';
 
+export interface Figure {
+  shape: FigureShape;
+  color: FigureColor;
+  getArea(): number;
 }
 
-export class Triangle {
+export function validate(...args: number[]): boolean {
+  const [a, b, c] = args;
+
+  if (args.length === 3) {
+    if (a < (b + c) && b < (a + c) && c < (a + b)) {
+      return false;
+    }
 
+    return true;
+  }
+
+  return args.some((side: number) => side <= 0);
```

Comment:
function is composed with different pieces of logic. Better to perform checks in place - in the constructors. Or separate into separate function for triangle.